spec/lib/guardian_extension_spec.rb

```diff
@@ -130,42 +130,92 @@
     describe "when translator enabled" do
       before { SiteSetting.translator_enabled = true }
 
-      describe "anon user" do
-        before { SiteSetting.restrict_translation_by_group = "#{Group::AUTO_GROUPS[:everyone]}" }
+      describe "when experimental_topic_translation enabled" do
+        before { SiteSetting.experimental_topic_translation = true }
 
-        it "cannot translate" do
-          expect(Guardian.new.can_translate?(post)).to eq(false)
+        describe "anon user" do
+          before { SiteSetting.restrict_translation_by_group = "#{Group::AUTO_GROUPS[:everyone]}" }
+
+          it "cannot translate" do
+            expect(Guardian.new.can_translate?(post)).to eq(false)
+          end
+        end
+
+        describe "logged in user" do
+          it "cannot translate when user is not in restrict_translation_by_group" do
+            SiteSetting.restrict_translation_by_group = "#{group.id + 1}"
+
+            expect(guardian.can_translate?(post)).to eq(false)
+          end
+
+          describe "user is in restrict_translation_by_group" do
+            before { SiteSetting.restrict_translation_by_group = "#{group.id}" }
+
+            describe "locale is :pt" do
+              before { I18n.stubs(:locale).returns(:pt) }
+
+              it "cannot translate when post detected locale matches i18n locale" do
+                post.set_detected_locale("pt")
+
+                expect(guardian.can_translate?(post)).to eq(false)
+              end
+
+              it "can translate when post's detected locale does not match i18n locale" do
+                post.set_detected_locale("jp")
+
+                expect(guardian.can_translate?(post)).to eq(true)
+              end
+
+              it "can translate when post does not have translation" do
```

Comment:
The setup for this test is exactly the same as the test before it. I think the set up for this test should be clearing out the post's translation?

Reply:
Thank you for clarifying 👍 Perhaps we can just combine the two test cases into one? Technically we can only translate when post's detected locale does not match i18n locale **and** also when the post does not have translation.

Might also be useful to assert that the post cannot be translated when the post's translation is present for the given locale.